core/commands/cmdenv/env.go

```diff
@@ -70,3 +71,28 @@ func GetConfigRoot(env cmds.Environment) (string, error) {
 
 	return ctx.ConfigRoot, nil
 }
+
+// EscNonPrint converts non-printable characters and backslash into Go
+// escape sequences, if the given string contains any.
+func EscNonPrint(s string) string {
+	// First see if escaping is needed, to avoid creating garbage.
```

Comment:
you mean garbage in the sense of memory that needs garbage collecting, right? Maybe just clarify that we're avoiding creating extraneous strings.

Reply:
IMO, best to simply remove the comment as code's intent is clear.